libs/cart/src/resolvers/empty-cart-resolver.service.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { TestBed, async } from '@angular/core/testing';
+import { StoreModule, combineReducers, Store } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { provideMockActions } from '@ngrx/effects/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { Router } from '@angular/router';
+
+import { DaffCart, DaffCartReducersState, daffCartReducers, DaffCartLoadSuccess }  from '@daffodil/cart';
+import { DaffCartFactory, DaffCartItemFactory } from '@daffodil/cart/testing';
+
+import { DaffEmptyCartResolver } from './empty-cart-resolver.service';
+import { DaffEmptyCartResolverRedirectUrl } from './tokens/empty-cart-resolver-redirect.token';
+import { DaffEmptyCartResolverModule } from './empty-cart-resolver.module';
+
+describe('DaffEmptyCartResolver', () => {
+  const actions$: Observable<any> = null;
+  let emptyCartResolver: DaffEmptyCartResolver;
+  let store: Store<DaffCartReducersState>;
+  let cartFactory: DaffCartFactory;
+  let cartItemFactory: DaffCartItemFactory;
+  let stubCart: DaffCart;
+	let router: Router;
+	const stubUrl = '/cart';
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        StoreModule.forRoot({
+          cart: combineReducers(daffCartReducers),
+        }),
+				RouterTestingModule,
+				DaffEmptyCartResolverModule
+      ],
+      providers: [
+				provideMockActions(() => actions$),
+				{ provide: DaffEmptyCartResolverRedirectUrl, useValue: stubUrl }
+      ]
+    });
+
+    emptyCartResolver = TestBed.get(DaffEmptyCartResolver);
+    cartFactory = TestBed.get(DaffCartFactory);
+    cartItemFactory = TestBed.get(DaffCartItemFactory);
+    stubCart = cartFactory.create();
+    store = TestBed.get(Store);
+    router = TestBed.get(Router);
+
+    spyOn(router, 'navigateByUrl');
+  }));
+
+  it('should be created', () => {
+    expect(emptyCartResolver).toBeTruthy();
+  });
+
+  describe('resolve', () => {
+
+    describe('when ResolverCartSuccessAction is dispatched with a cart', () => {
```

Comment:
This message doesn't quite seem correct. `ResolverCartSuccess` doesn't exist. Do you mean something like `when cart resolution suceeds`?

Reply:
oops. This was from copying over from demo. Fixed